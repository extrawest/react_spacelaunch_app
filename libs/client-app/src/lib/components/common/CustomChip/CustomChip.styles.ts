import type { SxProps, Theme } from '@mui/material';

import type { ChipStylesOptions } from './CustomChip.types';

const chip: (options: ChipStylesOptions) => SxProps<Theme> =
  ({ gradient }) =>
  (theme) => ({
    ...theme.typography.caption,
    height: 'auto',
    px: 3.75,
    py: 1.25,
    borderRadius: 10,
    background: gradient
      ? `linear-gradient(94.97deg, ${theme.palette.secondary.main} 3.92%, ${theme.palette.primary.main} 52.92%)`
      : theme.palette.primary.main,
  });

export const styles = {
  chip,
};
