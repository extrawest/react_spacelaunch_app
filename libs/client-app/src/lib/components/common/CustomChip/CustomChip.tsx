import type { FC } from 'react';

import { Chip } from '@mui/material';

import { styles } from './CustomChip.styles';
import type { CustomChipProps } from './CustomChip.types';

export const CustomChip: FC<CustomChipProps> = ({
  label,
  gradient = false,
}) => {
  return (
    <Chip
      label={label}
      sx={styles.chip({ gradient })}
    />
  );
};
