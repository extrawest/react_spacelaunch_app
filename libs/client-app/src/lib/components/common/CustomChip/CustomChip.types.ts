export interface CustomChipProps {
  label: string;
  gradient?: boolean;
}

export interface ChipStylesOptions {
  gradient: boolean;
}
