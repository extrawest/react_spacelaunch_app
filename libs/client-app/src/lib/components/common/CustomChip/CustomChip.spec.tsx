import { render } from '@testing-library/react';

import { CustomChip } from './CustomChip';

describe('[component]: CustomChip', () => {
  const label = 'Label';

  test('Should match default style', () => {
    const { baseElement } = render(<CustomChip label={label} />);

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Should match style with gradient', () => {
    const { baseElement } = render(
      <CustomChip
        label={label}
        gradient
      />
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Label should be correct', () => {
    const { getByText } = render(<CustomChip label={label} />);
    expect(getByText(label)).toHaveTextContent(label);
  });
});
