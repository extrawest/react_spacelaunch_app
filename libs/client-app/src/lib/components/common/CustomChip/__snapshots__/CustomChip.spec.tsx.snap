// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]: CustomChip Should match default style 1`] = `
<body>
  <div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-14f8vjg-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Label
      </span>
    </div>
  </div>
</body>
`;

exports[`[component]: CustomChip Should match style with gradient 1`] = `
<body>
  <div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-196zyfn-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Label
      </span>
    </div>
  </div>
</body>
`;
